### PR TITLE
use cmake for cashew

### PR DESCRIPTION
**Pull request by nickdesaulniers**
review? @kripken

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -1,29 +1,19 @@
...
-  proc = Popen(['sh', './build.sh'], stdout=PIPE)
-  proc.communicate()
-  assert proc.returncode == 0, 'failed to build'
+import os, difflib
+from subprocess import Popen, PIPE
 
 print 'testing'
 
-for i in os.listdir('samples'):
+for i in os.listdir('../samples'):
```
Comment (kripken) [inline test.py]: why is this change needed?

Comment (nickdesaulniers) [inline test.py]: if you build out of source, then the binary executable for cashew lives in `build/cashew` while samples is up one level.  If you built cashew in source, then this change would not be necessary.  That's why in the readme, I sugguest running test.py from `build/`.

**Comment by kripken** [comment]
Cool, thanks.
